src/ctia/task/migration/migrate_es_stores.clj

```diff
@@ -124,12 +124,14 @@
                :documents data
                :search_after next-search-after)))))
 
-(s/defn read-source ;; WARNING: defining schema output breaks lazyness
+(s/defn read-source
   "returns a lazy-seq of batch from source store"
   [read-params :- (s/maybe BatchParams)]
-  (lazy-seq
-   (when-let [batch (read-source-batch read-params)]
-     (cons batch (read-source (dissoc batch :documents))))))
+  (iteration read-source-batch
+             :vf identity
+             :initk read-params
+             :kf #(dissoc % :documents)
+             :somef seq))
```

Comment:
I think both `:vf` and `:somef` should be their default values. 

`:somef` => `some?` because it's testing a `(s/maybe BatchParams)`

`:vf` is already `identity`.